### PR TITLE
Add action to revive bumblebee previews.

### LINKED ISSUE #3619: Add view and button to revive the bumblebee preview.
Sometimes creating a bumblebee preview image/thumbnail/overlay image fails and in rare circumstances this leaves bumblebee in an inconsistent state (GEVER-side). Also due to a bug some cached bumblebee document checksums may not be up to date, see https://github.com/4teamwork/opengever.core/pull/3617 for example. It also may happen that the bumblebee service is down temporarily and the documents can't be registered.

At the moment these documents won't show a preview until the document is modified and thus triggers a bumblebee update. 

We would like to add a button to "revive" the preview as a stop-gap measure so that an `Administrator` or `Manager` can attempt to reset bumblebee state and make sure a preview image is rendered again.

"reviving" should perform the following actions:
- re-calculate bumblebee checksum
- store in bumblebee

can most likely be implemented by using something from`IBumblebeeDocument`.

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR adds the possibility to revive a bumblebee document.

screen2

See #3619 for more information.

closes #3619

### CODE REVIEW

File: opengever/bumblebee/browser/revive_preview.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            # Forces a document-update. Means, regenerating the checksum
+            # and queue storing in bumblebee.
+            IBumblebeeDocument(self.context)._handle_update(force=True)
+
+            api.portal.show_message(message=_(u'preview_revived',
+                                              default=u'Preview revived and will be available soon'),
+                                    request=self.request, type='info')
+        else:
+            api.portal.show_message(message=_(u'preview_revive_not_available',
+                                              default=u'Revive the preview is not available for this context.'),
```
Comment (deiferni) [inline opengever/bumblebee/browser/revive_preview.py]: Reviving

File: opengever/bumblebee/browser/revive_preview.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    This view lets you fix this inconsistent bumblebee document either thorugh
+    calling the view directly in the browser with
+
+    @@revive_preview/revive
+
+    or you use the action on the object to revive the current object.
+    """
+
+    def revive(self):
```
Comment (deiferni) [inline opengever/bumblebee/browser/revive_preview.py]: or just `__call__`? then calling this via browser would be easier as one has only to remember `revive_preview`.

File: opengever/bumblebee/browser/revive_preview.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        """
+        if self.available():
+
+            # Indicates that this context should not be converted if the
+            # checksum didn't change. Even the force attribute respects this
+            # marker-attribute.
+            #
+            # Because we really want to update the document, we just remove
+            # this property.
+            if hasattr(self.context, '_v_bumblebee_last_converted'):
```
[the remark this answers is not in the document]
Comment (elioschmutz) [reply opengever/bumblebee/browser/revive_preview.py]: From the Zope-Docs: https://docs.zope.org/zope2/zdgbook/ZODBPersistentComponents.html

> The third rule is that all object attributes that begin with ‘_v_’ are “volatile” and are not saved to the database. This means that as long as the persistent object is in Zope memory cache, volatile attributes can be used. When the object is deactivated (removed from memory) volatile attributes are thrown away.

I think thats the reason, why the volatile-attribute is still on the object. The object comes from the memory.

File: opengever/bumblebee/browser/revive_preview.py (hunk 2 above)
Comment (elioschmutz) [reply opengever/bumblebee/browser/revive_preview.py]: I thought `<property name="available_expr">object/@@revive_preview/available</property>` is calling the object first before calling the method... but I tested it and it doesn't 🤔 

So I changed it to revive directly on calling the object.x

File: opengever/bumblebee/browser/revive_preview.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    """Sometimes creating a bumblebee preview fails and in rare circumstances
+    this leaves bumblebee in an inconsistent state (GEVER-side).
+
+    It also may happen that the bumblebee service is down temporarily and the
+    documents can't be registered.
+
+    These documents won't show a preview until the document is modified and
+    thus triggers a bumblebee update.
+
+    This view lets you fix this inconsistent bumblebee document either thorugh
```
[the remark this answers is not in the document]
Comment (elioschmutz) [reply opengever/bumblebee/browser/revive_preview.py]: 👍

File: opengever/bumblebee/browser/revive_preview.py (hunk 1 above)
Comment (elioschmutz) [reply opengever/bumblebee/browser/revive_preview.py]: 👍

File: opengever/bumblebee/browser/revive_preview.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (deiferni) [reply opengever/bumblebee/browser/revive_preview.py]: 🤔  mhmm. It can't hurt to keep it here.

**Pull request merged**